src/components/SlateEditor/plugins/h5p/EditH5P.tsx

```diff
@@ -0,0 +1,177 @@
+import FormikFieldDescription from '../../../FormikField/FormikFieldDescription';
+
+const StyledModalBody = styled.div`
+  display: flex;
+  height: 100%;
+  flex-direction: column;
+  padding: 50px;
+  gap: 30px;
+`;
+
+const StyledModalContent = styled(ModalContent)`
+  padding: 0;
+  width: 100% !important;
+  height: 100%;
+  max-height: 95%;
+  overflow: hidden;
+`;
+
+const StyledMetadataModalContent = styled(ModalContent)``;
+
+const ButtonWrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: end;
+  gap: 10px;
+`;
+
+interface Props {
+  element: H5pElement;
+  embed: H5pMetaData | undefined;
+  isEditing: EditingState;
+  setIsEditing: (value: EditingState) => void;
+  language: string;
+  editor: Editor;
+}
+
+const EditH5P = ({ isEditing, setIsEditing, language, editor, element, embed }: Props) => {
+  const { t } = useTranslation();
+  const [alttext, setAlttext] = useState(embed?.embedData.alt);
+  const onSave = useCallback(
+    (params: OnSelectObject) => {
+      if (!params.path) {
+        return;
+      }
+      setIsEditing(undefined);
+      const cssUrl = encodeURIComponent(`${config.ndlaFrontendDomain}/static/h5p-custom-css.css`);
+      const url = `${config.h5pApiUrl}${params.path}?locale=${getH5pLocale(
+        language,
+      )}&cssUrl=${cssUrl}`;
+      const embedData: H5pEmbedData = {
+        resource: 'h5p',
+        path: params.path,
+        title: params.title,
+        alt: alttext,
+        url,
+      };
+      const properties = {
+        data: embedData,
+      };
+      ReactEditor.focus(editor);
+      const path = ReactEditor.findPath(editor, element);
+      Transforms.setNodes(editor, properties, { at: path });
+      if (Editor.hasPath(editor, Path.next(path))) {
+        setTimeout(() => {
+          Transforms.select(editor, Path.next(path));
+        }, 0);
+      }
+    },
+    [alttext, editor, element, language, setIsEditing],
+  );
+
+  const onSaveMetadata = useCallback(() => {
+    if (!embed?.embedData) {
+      return;
+    }
+
+    setIsEditing(undefined);
+    ReactEditor.focus(editor);
+    const data: H5pEmbedData = {
+      resource: 'h5p',
+      path: embed.embedData.path,
+      url: embed.embedData.url,
+      title: embed?.embedData.title,
+      alt: alttext,
+    };
+    Transforms.setNodes(editor, { data }, { at: ReactEditor.findPath(editor, element) });
+  }, [alttext, editor, element, embed?.embedData, setIsEditing]);
+
+  const onClose = () => {
+    setIsEditing(undefined);
+    ReactEditor.focus(editor);
+    const path = ReactEditor.findPath(editor, element);
+    if (Editor.hasPath(editor, Path.next(path))) {
+      setTimeout(() => {
+        Transforms.select(editor, Path.next(path));
+      }, 0);
+    }
+  };
+
+  const onChange = (e: ChangeEvent<HTMLTextAreaElement>) => setAlttext(e.target.value);
+
+  switch (isEditing) {
+    case 'metadata':
+      return (
+        <StyledMetadataModalContent size="small">
```

Comment:
Brukte samme hoved `<Modal>` element og endret innholdet basert på staten.
Er dette måten å gjøre det på? Føltes litt rart, men jeg syns hele den radix dialog tingen virket litt weird.

Reply:
Dette er nok ikke måten å gjøre det på, nei. Man kan gjøre det sånn her, men det er som oftest greiere å ha det som to forskjellige modaler.

Reply:
Hvordan gjør man det dersom man vil at triggerene skal ligge pent ved siden av hverandre "inni" modalen?
Trenger ikke `Root` å wrappe hele greia?

Reply:
Da tror jeg nok du må wrappe hver knapp i sin egen `Modal`-komponent. Altså at begge er frittstående. Det portalles ut, og bør ikke påvirke layout.